### DISCUSSION TITLE
How to center window on screen and fit to contents?

### DISCUSSION THREAD

**Discussion by GirkovArpa** [c-smile/sciter-js-sdk#39] [Q&A] [answered]
Is there a Sciter.JS equivalent?  It's the only way I know how to center the window and also fit it to its contents, in TIScript:
```javascript
function self.ready() {
  const w = self.intrinsicWidthMax();
  const h = self.intrinsicHeight();
  const (sw, sh) = view.screenBox(#frame, #dimension);
  view.move((sw / 2) - (w / 2), (sh / 2) - (h / 2), w, h, true);
}
```

**Comment by c-smile** [comment, marked as answer]
Pretty much as it is:

```JavaScript
const [wmin,w] = document.state.contentWidths();
const h = document.state.contentHeight(w);

const [sw,sh] = Window.this.screenBox("frame", "dimension");

Window.this.move( (sw - w) / 2, (sh - h) / 2, w, h, true );
```

Relevant documentation:  [document/element.state](https://github.com/c-smile/sciter-js-sdk/blob/main/docs/md/Element.State.md)  and [Window.this](https://github.com/c-smile/sciter-js-sdk/blob/main/docs/md/Window.md)

**Reply by GirkovArpa** [reply]
Fantastic, thank you!

**Reply by liudonghua123** [reply]
Thanks, add a note, you need to use `scapp` from `sciter-js-sdk` instead of `sciter-sdk`. 😄
